Bottom: 10ecf8b7802e5761976e9a821167199fbf21756f
Top:    52c3498efeb997fb321727c72f7774fcb7116e46
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-27 01:40:34 +0100

Create new function GetLayerGroupNumberBySide()


---

diff --git a/src/misc.c b/src/misc.c
index 8d12706..de94eb9 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1440,6 +1440,19 @@ GetLayerGroupNumberByNumber (Cardinal Layer)
 }
 
 /* ---------------------------------------------------------------------------
+ * returns the layergroup number for the passed side (TOP_LAYER or BOTTOM_LAYER)
+ */
+int
+GetLayerGroupNumberBySide (int side)
+{
+  /* Find the relavant board side layer group by determining the
+   * layer group associated with the relevant side's silk-screen
+   */
+  return GetLayerGroupNumberByNumber(
+      side == TOP_SIDE ? component_silk_layer : solder_silk_layer);
+}
+
+/* ---------------------------------------------------------------------------
  * returns a pointer to an objects bounding box;
  * data is valid until the routine is called again
  */
diff --git a/src/misc.h b/src/misc.h
index 9ce0905..25db4dc 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -74,6 +74,7 @@ void SaveOutputWindow (void);
 int GetLayerNumber (DataType *, LayerType *);
 int GetLayerGroupNumberByPointer (LayerType *);
 int GetLayerGroupNumberByNumber (Cardinal);
+int GetLayerGroupNumberBySide (int);
 int GetGroupOfLayer (int);
 int ChangeGroupVisibility (int, bool, bool);
 void LayerStringToLayerStack (char *);
